src/docs/tool-libfuzzer.dox

```diff
@@ -0,0 +1,210 @@
+/*! @page tool-libfuzzer Testing with LLVM LibFuzzer
+
+# Building and running an LLVM LibFuzzer target
+
+LLVM LibFuzzer is in-process, coverage-guided, evolutionary fuzzing engine. It feeds a series of
+fuzzed inputs via a "target" function and attempts to trigger crashes, memory bugs and undefined
+behavior.
+
+## Step 1: Select Clang as your C compiler in your build configuration
+
+Support for LibFuzzer is implemented as a compiler flag in Clang. Our configuration checks whether
+the compiler in use supports \c -fsanitize=fuzzer and if so, elects to build the fuzz targets.
+
+Compiling with the ASan isn't mandatory but it is recommended since fuzzing often exposes memory
+bugs.
+
+@code
+$ cd build_posix/
+$ ../configure CC="clang-8" CFLAGS="-fsanitize=address"
+@endcode
+
+## Step 2: Build as usual
+
+@code
+$ cd build_posix
+$ make
+@endcode
+
+## Step 3: Run a fuzz target
+
+Each fuzz target is just a program under \c build_posix/test/fuzz/. Once you become familiar with
+LibFuzzer, it may be easier to run the target program directly, but for first-time users, the script
+at \c test/fuzz/fuzz_run.sh is handy. It performs a limited number of runs, automatically cleans up
+after itself in between runs and provides a sensible set of parameters which you can add to. For
+example:
+
+@code
+$ cd build_posix/test/fuzz/
+$ bash ../../../test/fuzz/fuzz_run.sh ./fuzz_modify
+@endcode
+
+In general the usage is:  
+
+@code
+fuzz_run.sh <fuzz-test-binary> [fuzz-test-args]
+@endcode
+
+Each fuzz target will produce a few outputs:
+
+- \c fuzz-N.log: The LibFuzzer log for worker N.
+
+- \c WT_TEST_<pid>: The home directory for a given worker process.
+
+- \c crash-<input-hash>: If an error occurs, you may see a file like this containing the input that
+crashed the target.
+
+- \c WT_TEST_<pid>.profraw: If you're running with Clang coverage (more on this later), you'll see
+files containing profiling data for a given worker. These will be used by \c fuzz_coverage.
+
+### Corpus
+
+LibFuzzer is a coverage based fuzzer meaning that it notices when a particular input hits a new code
+path and adds it to a "corpus" of interesting data inputs. It then uses existing data in the corpus
+to mutate and come up with new inputs.
+
+While LibFuzzer will automatically add to its corpus when it finds an interesting input, some fuzz
+targets (especially those that expect data in a certain format) require a corpus to start things off
+in order to be effective. The target \c fuzz_config is one example of this as it expects its data
+sliced with a separator so the fuzzer needs some examples to guide it. The corpus is supplied as the
+first positional argument to both \c fuzz_run.sh and the underlying fuzz target itself. For example:
+
+@code
+$ cd build_posix/test/fuzz/
+$ bash ../../../test/fuzz/fuzz_run.sh ./fuzz_config ../../../test/fuzz/config/corpus/
+@endcode
```

Comment:
I'm still figuring out how this is going to work. I think in MongoDB, Evergreen manages the corpus and people don't really run the fuzzer locally unless it's to reproduce a failure.

So the idea is that when a failure happens in Evergreen, you just grab the input that failed and run it locally. You don't really need the huge Evergreen corpus since you're not generating inputs yourself.

I'm just checking in the corpus for `fuzz_config` since it makes it more useable but I'm not sure it belongs here in the long term.